Tests: Add and modify testcases for pipe.

youhuazhu opened the pull request:
1. Add a new testcase for defining and initializing pipes at run time.
2. Add detail annotations for some testcases.

Signed-off-by: Zhu YouhuaX <youhuax.zhu@intel.com>

youhuazhu committed:
Tests: Add and modify testcases for pipe.

1. Add a new testcase for defining and initializing pipes at run time.
2. Add details comments for some testcases.

Signed-off-by: Zhu YouhuaX <youhuax.zhu@intel.com>

youhuazhu commented:
@mm387561

andyross commented on tests/kernel/pipe/pipe/src/test_pipe.c:
```diff
@@ -874,3 +880,81 @@ void test_pipe_put_min_xfer(void)
+ * @ingroup kernel_pipe_tests
+ * @see K_PIPE_DEFINE()
+ */
+void test_pipe_define_at_compileTime(void)
```
Call this "test_static_pipe" for clarity, or possibly "test_k_pipe_define".  Also interCapsAreNotZephyrStyle.

youhuazhu replied:
Hi andyross, thanks for quickly response,  yes,  I found that have the related case in this test suite,  and I removed this one.

andyross commented on tests/kernel/pipe/pipe/src/test_pipe.c:
```diff
@@ -874,3 +880,81 @@ void test_pipe_put_min_xfer(void)
+ * - Using the pipe which defined and initialized at compile time to
+ * transfer data and verify it.
+ * @ingroup kernel_pipe_tests
+ * @see K_PIPE_DEFINE()
```
Uh... where is the K_PIPE_DEFINE() for this pipe?  If it was already in the code, then surely this test case was already covered?

youhuazhu replied:
yes,  already covered by other test case,  thanks for pointing out.

andyross commented on tests/kernel/pipe/pipe/src/test_pipe.c:
```diff
@@ -874,3 +880,81 @@ void test_pipe_put_min_xfer(void)
+
+	/* get test_pipe data if something there */
+	k_pipe_get(&test_pipe, &rx_buffer, PIPE_SIZE, &read,
+			NO_CONSTRAINT, K_NO_WAIT);
```
This doesn't seem like much of a test.  It's ignoring the return value and dropping the "read" output on the floor without checking.  I'm guessing you want to check the output for -EAGAIN at least?

youhuazhu replied:
I have removed this case due to the case covered by others.

andyross reviewed (commented):
These seem really simple, and I can't imagine these code paths were actually untested.  Aren't we doing effectively the same thing in some other test in this suite already?  Maybe you can remove those cases while you're here?

What was the code coverage or whatever result that inspired this work?  It seems like it's probably something that could have been better addressed by one or two extra lines in an existing test.  Adding test code just for the sake of adding code bloats the test suite and slows down CI for everyone.

youhuazhu commented:
Hi @andyross , I have updated my pr according your comments, would you help to review it again:  added a new test case for  defining and initializing pipes at run time which not covered by other cases.
thank you.

andyross commented on tests/kernel/pipe/pipe/src/test_pipe.c:
```diff
@@ -874,3 +883,41 @@ void test_pipe_put_min_xfer(void)
+ */
+void test_pipe_define_at_runtime(void)
+{
+	unsigned char __aligned(4) ring_buffer[PIPE_SIZE];
```
What does this need alignment?  If it's because the test code is using the buffer as an array of other types, then you should declare the memory as an array of whatever the natural type is.  If it's because the pipe code is doing it and hitting bus errors at runtime, we have a bug in the pipe layer (which is supposed to be byte-aligned only).  If it's neither (and I think it's neither), just remove the alignment restriction.

youhuazhu replied:
o,  yes,  needn't memory alignment here,  Just moved,  thanks for reminding. :)

andyross commented on tests/kernel/pipe/pipe/src/test_pipe.c:
```diff
@@ -874,3 +883,41 @@ void test_pipe_put_min_xfer(void)
+	zassert_true(rx_buffer_check(rx_buffer, read) == read,
+			"Bytes read are not match.");
+
+	ztest_test_pass();
```
Remove this line.  Returning from a top-level ztest function without having flagged a failure is a "pass" result, by definition.

youhuazhu replied:
@andyross ,  Have removed the pass line,  thank you for reviewing again.

andyross reviewed (commented):
Two nitpicks.  Nothing that looks wrong.

youhuazhu commented:
Hi @andyross,  Would you help to review my this pr again?  thanks.

andyross reviewed (approved):
Looks good.  Thanks.

youhuazhu commented:
thanks andyross for reviewing,  
Hi @nashif @andrewboie , would you help to review this pr?  thanks.

youhuazhu commented:
Have rebased this PR, and re-verified.
